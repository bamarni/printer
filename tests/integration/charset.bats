#!/usr/bin/env bats

@test "[Charset] print some charset-specific text" {
  cmp <(printer < $BATS_TEST_DIRNAME/fixtures/charset 2>/dev/null) $BATS_TEST_DIRNAME/fixtures/charset.expected
}

@test "[Charset] fail for unsupported charsets" {
  run printer < $BATS_TEST_DIRNAME/fixtures/charset_unexisting

  [[ "$output" == *"Charset TOUTI not supported"* ]] || false

  [ "$status" -ne 0 ]
}

@test "[Charset] fail for unencodable unicode" {
  run printer < $BATS_TEST_DIRNAME/fixtures/charset_unencodable

  [[ "$output" == *"Couldn't encode to charset"* ]] || false

  [ "$status" -ne 0 ]
}
