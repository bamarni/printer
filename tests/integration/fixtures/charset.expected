tÒtre ou ne pas ˆtre
Òtre ou ne pas ˆtre